piker/brokers/kraken.py

```diff
@@ -271,18 +446,333 @@ async def bars(
             raise SymbolNotFound(json['error'][0] + f': {symbol}')
 
 
+
 @asynccontextmanager
 async def get_client() -> Client:
-    client = Client()
+
+    section = get_config()
+    client = Client(
+        name=section['key_descr'],
+        api_key=section['api_key'],
+        secret=section['secret']
+    )
 
     # at startup, load all symbols locally for fast search
     await client.cache_symbols()
 
     yield client
 
 
-async def stream_messages(ws):
+def pack_positions(
+    acc: str,
+    trades: dict
+) -> list[Any]:
+    positions: dict[str, float] = {}
+    vols: dict[str, float] = {}
+    costs: dict[str, float] = {}
+    position_msgs: list[Any] = []
+
+    for trade in trades.values():
+        sign = -1 if trade['type'] == 'sell' else 1
+        # This catch is for populating the dict with new values
+        # as the plus assigment will fail if there no value
+        # tied to the key
+        pair = trade['pair']
+        vol = float(trade['vol'])
+        # This is for the initial addition of a pair so the 
+        # += operation does not fail.
+        vols[pair] = vols.setdefault(pair, 0)
```

Comment:
Sorry why are you doing what seems to be a needless assignment here?

Reply:
In fact the more terse way to do this would be to avoid the `+=` syntax and drop the half the lines:

```python
vols[pair] = vols.setdefault(pair, 0) + sign * vol
```

Reply:
heh, actually come to think of it, if you *must* do an assignment then you don't really need `.setdefault()`; just use `vols.get(pair, 0)` instead in the first term.

Reply:
yea I like the `.get(pair, 0)` approach better. will get on it